packages/rum-core/src/boot/startRum.ts

```diff
@@ -178,6 +179,11 @@ export function startRum(
     urlContexts
   )
 
+  const startDurationVital = (startVital: DurationVitalStart) =>
+    createVitalInstance((vital) => {
+      vitalCollection.addDurationVital(vital)
+    }, startVital)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:**  This code could be collocated in vitalCollection module. For instance something like
```ts
export function startVitalCollection(lifeCycle: LifeCycle, pageStateHistory: PageStateHistory) {
  function isValid(vital: DurationVital) {
    return !pageStateHistory.wasInPageStateDuringPeriod(PageState.FROZEN, vital.startClocks.relative, vital.duration)
  }

  function addDurationVital(vitalAdd: DurationVitalAdd) {
    const vital = assign({ type: VitalType.DURATION }, vitalAdd)

    if (isValid(vital)) {
      lifeCycle.notify(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, processVital(vital, true))
    }
  }

  return {
    addDurationVital,
    startDurationVital: (startVital: DurationVitalStart) =>
      createVitalInstance((vital) => {
        addDurationVital(vital)
      }, startVital),
  }
}
```
‏

Reply:
That's what I planned. But I has extracted the logic to be able to provide an implementation for `preStartRum`:

```typescript
return createVitalInstance((vital) => {
   bufferApiCalls.add((startRumResult) => startRumResult.addDurationVital(vital))
}, vitalStart)
```

This API function is specific because it returns the vital instance on which can call `stop` (it calls the callback).

Reply:
I think it is reasonable to have the preStart implementation in preStart and the postStart in vitalCollection. Wdyt?

Reply:
💯 make sense